### PR TITLE
p2p: Stop peers gracefully when stopping switch

### PR DESCRIPTION
**Pull request by SebastianElvis**
This PR fixed `TODO: gracefully disconnect from peers.` in `p2p/switch.go`.

* [ ] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: p2p/switch.go [hunk 1]
```diff
@@ -221,11 +221,8 @@ func (sw *Switch) OnStart() error {
 func (sw *Switch) OnStop() {
 	// Stop peers
 	for _, p := range sw.peers.List() {
-		sw.transport.Cleanup(p)
-		p.Stop()
-		if sw.peers.Remove(p) {
-			sw.metrics.Peers.Add(float64(-1))
-		}
+		sw.StopPeerGracefully(p)
+		sw.metrics.Peers.Add(float64(-1))
```
Comment (melekes) [inline p2p/switch.go]: `StopPeerGracefully` already decrements this counter

File: p2p/switch.go [hunk 2]
```diff
@@ -221,11 +221,8 @@ func (sw *Switch) OnStart() error {
 func (sw *Switch) OnStop() {
 	// Stop peers
 	for _, p := range sw.peers.List() {
-		sw.transport.Cleanup(p)
-		p.Stop()
-		if sw.peers.Remove(p) {
-			sw.metrics.Peers.Add(float64(-1))
-		}
+		sw.StopPeerGracefully(p)
```
Comment (melekes) [inline p2p/switch.go]: I think it's better to call `sw.stopAndRemovePeer(peer, nil)` to avoid excessive logging

File: p2p/switch.go (hunk 1 above)
Comment (SebastianElvis) [reply p2p/switch.go]: resolved

File: p2p/switch.go (hunk 2 above)
Comment (SebastianElvis) [reply p2p/switch.go]: resolved

**Comment by SebastianElvis** [comment]
> 👍
> 
> Although I am not sure this PR actually changes anything except removing duplicated code. I think we already do "gracefully disconnect from peers", so the TODO can be removed.

Yeah, it seems that the original code works.
Anyway, this PR can remove some duplicated code.
I have addressed your change requests. How about now?

**Pull request merged**